src/backends/torch/torchinputconns.h

```diff
@@ -0,0 +1,109 @@
+/**
+ * DeepDetect
+ * Copyright (c) 2018 Jolibrain
+ * Author: Louis Jean <ljean@etud.insa-toulouse.fr>
+ *
+ * This file is part of deepdetect.
+ *
+ * deepdetect is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * deepdetect is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with deepdetect.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#ifndef TORCHINPUTCONNS_H
+#define TORCHINPUTCONNS_H
+
+#include "imginputfileconn.h"
+#include "txtinputfileconn.h"
+
+#include <vector>
+
+namespace dd
+{
+    class TorchInputInterface
+    {
+    public:
+      TorchInputInterface() {}
+      TorchInputInterface(const TorchInputInterface &i)
+      {
+
+      }
+
+      ~TorchInputInterface() {}
+    };
+
+    class ImgTorchInputFileConn : public ImgInputFileConn, public TorchInputInterface
+    {
+    public:
+        ImgTorchInputFileConn()
+            :ImgInputFileConn() {}
+        ImgTorchInputFileConn(const ImgTorchInputFileConn &i)
+            :ImgInputFileConn(i),TorchInputInterface(i) {}
+        ~ImgTorchInputFileConn() {}
+
+        // for API info only
+        int width() const
+        {
+            return _width;
+        }
+
+        // for API info only
+        int height() const
+        {
+            return _height;
+        }
+
+        void init(const APIData &ad)
+        {
+            ImgInputFileConn::init(ad);
+        }
+        
+        void transform(const APIData &ad)
+        {
+            try
+            {
+                ImgInputFileConn::transform(ad);
+            }
+            catch(const std::exception& e)
+            {
+                throw;
+            }
+
+            std::vector<at::Tensor> tensors;
+
+            for (const cv::Mat &bgr : this->_images) {
+                _height = bgr.rows;
```

Comment:
`width` and `height` are usually set at service creation via the options of the input connector. Since they are class variables we may not want to set them up again here within the loop.